vault/hcp_link/capabilities/meta/meta.go

```diff
@@ -189,3 +191,110 @@ func (h *hcpLinkMetaHandler) ListAuths(ctx context.Context, req *meta.ListAuthsR
 		Auths: auths,
 	}, nil
 }
+
+func (h *hcpLinkMetaHandler) GetClusterStatus(ctx context.Context, req *meta.GetClusterStatusRequest) (*meta.GetClusterStatusResponse, error) {
+	if h.wrappedCore.HAState() != consts.Active {
+		return nil, fmt.Errorf("node not active")
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+
+	haEnabled := h.wrappedCore.HAEnabled()
+	haStatus := &meta.HAStatus{
+		Enabled: haEnabled,
+	}
+
+	if haEnabled {
+		leader := &meta.HANode{
+			Hostname: hostname,
+		}
+
+		peers := h.wrappedCore.GetHAPeerNodesCached()
+
+		haNodes := make([]*meta.HANode, len(peers)+1)
+		haNodes[0] = leader
+
+		for i, peerNode := range peers {
+			haNodes[i+1] = &meta.HANode{
+				Hostname: peerNode.Hostname,
+			}
+		}
+
+		haStatus.Nodes = haNodes
+	}
+
+	raftStatus := &meta.RaftStatus{}
+	raftConfig, err := h.wrappedCore.GetRaftConfiguration(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if raftStatus != nil {
+		raftServers := make([]*meta.RaftServer, len(raftConfig.Servers))
```

Comment:
It seems like this panic if `raftConfig` is nil. What do you think?

Reply:
Fixed https://github.com/hashicorp/vault/pull/18351/commits/c8e012f07d44d5bd5b3af9e03f7ecb2de549d59c